django/db/backends/base/schema.py

```diff
@@ -459,15 +460,22 @@ def add_field(self, model, field):
             constraint_suffix = '_fk_%(to_table)s_%(to_column)s'
             # Add FK constraint inline, if supported.
             if self.sql_create_column_inline_fk:
+                fk_constraint_name = self._fk_constraint_name(model, field, constraint_suffix)
                 to_table = field.remote_field.model._meta.db_table
                 to_column = field.remote_field.model._meta.get_field(field.remote_field.field_name).column
                 definition += " " + self.sql_create_column_inline_fk % {
-                    'name': self._fk_constraint_name(model, field, constraint_suffix),
+                    'name': fk_constraint_name,
                     'column': self.quote_name(field.column),
                     'to_table': self.quote_name(to_table),
                     'to_column': self.quote_name(to_column),
                     'deferrable': self.connection.ops.deferrable_sql()
                 }
+                # Setting the constraint to IMMEDIATE to allow changing data in
+                # the same transaction, if required.
+                if self.sql_column_inline_fk_immediate:
+                    self.deferred_sql.append(self.sql_column_inline_fk_immediate % {
```

Comment:
It feels like this should be run immediately after the column addition at `self.execute(sql, params)` and not deferred like it is right now.

Reply:
Agreed, I added a new test for this scenario.